### ISSUE TITLE
Raise error if target boxes are degenerate in Faster R-CNN

### ISSUE THREAD

**Issue by fmassa** [pytorch/vision#2240]
We have had a number of reports with users saying that their training loss is nan after a few iterations.

Most of the time, this is due to degenerate boxes (i.e., boxes with negative sizes or zero area). We should improve the user experience in those situations.

I think that raising an error in `GeneralizedRCNN` if the target boxes are degenerate would be a good compromise.

Related issues: https://github.com/pytorch/vision/issues/2235 https://github.com/pytorch/vision/issues/1994 https://github.com/pytorch/vision/issues/1176 https://github.com/pytorch/vision/issues/1128 #1120 and #997

**Comment by nvs-abhilash** [comment]
@fmassa which option would be better, to Raise an Exception or to print a warning and do something like you suggested here: [https://github.com/pytorch/vision/issues/2235#L82](https://github.com/pytorch/vision/issues/2235#L82) ?

**Comment by fmassa** [comment]
I'm leaning towards raising an error, so that it is explicit to the user that there is a problem with their data and they should fix it.

**Comment by fmassa** [comment]
Fixed in https://github.com/pytorch/vision/pull/2258
